pkg/keycheck/keycheck_test.go

```diff
@@ -0,0 +1,149 @@
+package keycheck
+
+import "testing"
+
+func Test_IsRandomKey(t *testing.T) {
+	tests := []struct {
+		name string
+		key  string
+		want bool
+	}{
+		{
+			name: "Valid UUID",
+			key:  "123e4567-e89b-12d3-a456-426614174000",
+			want: true,
+		},
+		{
+			name: "Valid Hex String",
+			key:  "abcdef1234567890abcdef1234567890",
+			want: true,
+		},
+		{
+			name: "Valid Base64 String",
+			key:  "QWxhZGRpbjpvcGVuIHNlc2FtZQ==",
+			want: true,
+		},
+		{
+			name: "Long String Without Vowels",
+			key:  "bcdfghjklmnpqrstvwxyz",
+			want: true,
+		},
+		{
+			name: "Mixed Case with Digits",
+			key:  "Abcdef123456",
+			want: false,
+		},
+		{
+			name: "Short String",
+			key:  "short",
+			want: false,
+		},
+		{
+			name: "String with Vowels",
+			key:  "aeiouAEIOU",
+			want: false,
+		},
+		{
+			name: "Mixed Case without Digits",
+			key:  "Abcdefghijklmnop",
+			want: true,
+		},
```

Comment:
This could be a problem because data from AWS could easily be like this (they use CapitalCase, and keys can be lengthy)

Reply:
hmm, so 256 length limit should just be good right ?

Reply:
By lengthy, i mean long enough to trigger the following logic.

```
			case len(segment) > 12 && !containsVowels(segment):
				return true
			case len(segment) > 16 && hasUpperLowerDigit(segment):
				return true

```

Reply:
is around 30 a good number for a segment then ?

Reply:
Yes, that sounds good to me.